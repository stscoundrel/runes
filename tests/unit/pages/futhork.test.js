import ReactDOM from 'react-dom'
import Futhork from 'pages/futhork'
import renderer from 'react-test-renderer'

describe('Futhork runic keyboard page', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Futhork />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<Futhork />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
