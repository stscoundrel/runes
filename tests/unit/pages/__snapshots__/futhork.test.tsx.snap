// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Futhork runic keyboard page Matches snapshot 1`] = `
Array [
  <main
    className="container"
  >
    <section>
      <header>
        <h1>
          Futhork
        </h1>
        <p>
          Translate runes to text
        </p>
      </header>
      <p
        className="result"
      />
      <div
        className="keyboard"
      >
        <button
          aria-label="Runic key ᚠ, meaning letter f"
          className="key"
          onClick={[Function]}
        >
          ᚠ
        </button>
        <button
          aria-label="Runic key ᚢ, meaning letter u"
          className="key"
          onClick={[Function]}
        >
          ᚢ
        </button>
        <button
          aria-label="Runic key ᚦ, meaning letter þ"
          className="key"
          onClick={[Function]}
        >
          ᚦ
        </button>
        <button
          aria-label="Runic key ᚮ, meaning letter o"
          className="key"
          onClick={[Function]}
        >
          ᚮ
        </button>
        <button
          aria-label="Runic key ᚱ, meaning letter r"
          className="key"
          onClick={[Function]}
        >
          ᚱ
        </button>
        <button
          aria-label="Runic key ᚴ, meaning letter k"
          className="key"
          onClick={[Function]}
        >
          ᚴ
        </button>
        <button
          aria-label="Runic key ᚼ, meaning letter h"
          className="key"
          onClick={[Function]}
        >
          ᚼ
        </button>
        <button
          aria-label="Runic key ᚿ, meaning letter n"
          className="key"
          onClick={[Function]}
        >
          ᚿ
        </button>
        <button
          aria-label="Runic key ᛁ, meaning letter i"
          className="key"
          onClick={[Function]}
        >
          ᛁ
        </button>
        <button
          aria-label="Runic key ᛆ, meaning letter a"
          className="key"
          onClick={[Function]}
        >
          ᛆ
        </button>
        <button
          aria-label="Runic key ᛌ, meaning letter s"
          className="key"
          onClick={[Function]}
        >
          ᛌ
        </button>
        <button
          aria-label="Runic key ᛋ, meaning letter s"
          className="key"
          onClick={[Function]}
        >
          ᛋ
        </button>
        <button
          aria-label="Runic key ᛐ, meaning letter t"
          className="key"
          onClick={[Function]}
        >
          ᛐ
        </button>
        <button
          aria-label="Runic key ᛏ, meaning letter t"
          className="key"
          onClick={[Function]}
        >
          ᛏ
        </button>
        <button
          aria-label="Runic key ᛒ, meaning letter b"
          className="key"
          onClick={[Function]}
        >
          ᛒ
        </button>
        <button
          aria-label="Runic key ᛘ, meaning letter m"
          className="key"
          onClick={[Function]}
        >
          ᛘ
        </button>
        <button
          aria-label="Runic key ᛚ, meaning letter l"
          className="key"
          onClick={[Function]}
        >
          ᛚ
        </button>
        <button
          aria-label="Runic key ᛦ, meaning letter y"
          className="key"
          onClick={[Function]}
        >
          ᛦ
        </button>
        <button
          aria-label="Runic key :, meaning letter  "
          className="key"
          onClick={[Function]}
        >
          :
        </button>
        <button
          aria-label="Runic key ᚵ, meaning letter g"
          className="key"
          onClick={[Function]}
        >
          ᚵ
        </button>
        <button
          aria-label="Runic key ᚽ, meaning letter e"
          className="key"
          onClick={[Function]}
        >
          ᚽ
        </button>
        <button
          aria-label="Runic key ᚯ, meaning letter ø"
          className="key"
          onClick={[Function]}
        >
          ᚯ
        </button>
        <button
          aria-label="Runic key ᛅ, meaning letter æ"
          className="key"
          onClick={[Function]}
        >
          ᛅ
        </button>
        <button
          aria-label="Runic key ᚰ, meaning letter ǫ"
          className="key"
          onClick={[Function]}
        >
          ᚰ
        </button>
        <button
          aria-label="Runic key ᛕ, meaning letter ᴘ"
          className="key"
          onClick={[Function]}
        >
          ᛕ
        </button>
      </div>
    </section>
  </main>,
  <footer
    className="section"
  >
    <div
      className="container text-center"
    >
      <small
        className="copyright"
      >
        Copyright © 
        2021
        <a
          href="https://github.com/stscoundrel"
          rel="noopener noreferrer"
          target="_blank"
        >
          Sampo Silvennoinen / StScoundrel
        </a>
      </small>
    </div>
  </footer>,
]
`;
